src/main/java/catbot/CatBot.java

```diff
@@ -0,0 +1,102 @@
+import catbot.io.ErrorIndicatorIo;
+import catbot.io.UserIo;
+import catbot.task.Task;
+import catbot.task.TaskList;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+
+public class CatBot {
+
+    private static final CommandMap commands = new CommandMap();
+    static final UserIo io = new CatbotConsoleIO();
+    private static final TaskList taskList = new TaskList("Tasks.txt");
+    static {
+        addSupportedCommandsToCommandMap();
+    }
+
+    private static void addSupportedCommandsToCommandMap() {
+        CommandPattern<Integer> integerPattern = CatBotCommandPatterns.getIntegerPatternGenerator()
+                .generate(io::indicateInvalidInteger);
+        CommandPattern<NamedParameterMap> slashPattern = CatBotCommandPatterns.getSlashPatternGenerator()
+                .generate(CatBotCommandPatterns.NO_DEFAULT);
+
+        commands.setDefaultCommand(io::indicateInvalidCommand)
+                .addCommand("bye", args -> io.cleanup())
+                .addCommand("list", args -> io.printTaskList(taskList));
+
+        // User modifying existing tasks (through IntegerPattern, and Index)
+        BiConsumer<String, Consumer<Integer>> runIfValidIndexElseIndicateError =
+                (args, lambda) -> integerPattern.ifParsableElseDefault(args,
+                        integer -> taskList.ifValidIndexElse(integer,
+                                lambda,
+                                invalidIndex -> io.indicateInvalidIndex(invalidIndex, taskList.getIndexBounds())
+                        ));
+
+                //noinspection SpellCheckingInspection
+        commands.addCommand("mark",
+                        string -> runIfValidIndexElseIndicateError.accept(string,
+                                validIndex -> {
+                                    taskList.markTask(validIndex-1);
+                                    io.printTaskModified(taskList, validIndex);
+                                }
+                        )
+                )
+                .addCommand("unmark",
+                        string -> runIfValidIndexElseIndicateError.accept(string,
+                                validIndex -> {
+                                    taskList.unmarkTask(validIndex-1);
+                                    io.printTaskModified(taskList, validIndex);
+                                }
+                        )
+                )
+                .addCommand("delete",
+                        string -> runIfValidIndexElseIndicateError.accept(string,
+                                validIndex -> io.printTaskDeleted(taskList.removeTask(validIndex-1))
+                        )
+                );
+
+        // User creating new tasks (with SlashPattern)
+
+        BiConsumer<String, BiFunction<NamedParameterMap, BiConsumer<ErrorIndicatorIo.InvalidState, NamedParameterMap>, Optional<Task>>>
```

Comment:
Perhaps u can split this line into 2? I think this line has 135 characters which exceeds the 120 characters limit stated in the coding standard.

Reply:
Got it, and thanks for all the nudges for coding style. 
I ended up running it through a stylechecker (finally), but I appreciate the pointers!